tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_swap_elements_kk.hpp

```diff
@@ -0,0 +1,94 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SWAP_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SWAP_HPP_
+
+#include <utility>
+
+namespace KokkosKernelsSTD {
+
+// this is here until we can use kokkos 3.6 which has swap avail
+template <class T>
+KOKKOS_INLINE_FUNCTION void _my_tmp_swap(T& a, T& b) noexcept {
+  static_assert(
+      std::is_move_assignable<T>::value &&
+      std::is_move_constructible<T>::value,
+      "KokkosKernelsSTD::swap arguments must be move assignable and move constructible");
+
+  T tmp = std::move(a);
+  a     = std::move(b);
+  b     = std::move(tmp);
+}
+
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
+	 class ElementType_x,
+         std::experimental::extents<>::size_type ... ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ... ext_y,
+         class Layout_y>
+  requires (sizeof...(ext_x) == sizeof...(ext_y))
+void swap_elements(kokkos_exec<ExeSpace> /*kexe*/,
+		   std::experimental::mdspan<
+		     ElementType_x,
+		     std::experimental::extents<ext_x ...>,
+		     Layout_x,
+		     std::experimental::default_accessor<ElementType_x>
+		   > x,
+		   std::experimental::mdspan<
+		     ElementType_y,
+		     std::experimental::extents<ext_y ...>,
+		     Layout_y,
+		     std::experimental::default_accessor<ElementType_y>
+		   > y)
+{
+  // constraints
```

Comment:
"Constraints" are implemented through SFINAE, concepts, or `requires` clauses.  "Mandates" are implemented through `static_assert` or something like it.